engine/src/main/java/io/camunda/zeebe/engine/state/migration/JobBackoffRestoreMigration.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.state.migration;
+
+import io.camunda.zeebe.engine.state.immutable.ProcessingState;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessingState;
+
+public class JobBackoffRestoreMigration implements MigrationTask {
+
+  @Override
+  public String getIdentifier() {
+    return getClass().getSimpleName();
+  }
+
+  @Override
+  public boolean needsToRun(final ProcessingState processingState) {
+    // The migration should run only if there are failed jobs in the state, but the backoff column
+    // family is empty or there are more failed jobs then record inside the backoff column family
+    return processingState.getJobState().isJobBackoffToRestore();
```

Comment:
❌ I think we should make this a bit more clever, and not iterate over all failed jobs, it is enough to find one and directly return. So maybe a method which checks exist at least one failed job.

Reply:
In this way we can cover more cases, do you think is enough to run the migration if the backoff cf is empty and there is at least one failed job ? 
Iterate on the failed jobs ensure that the migration runs only if there are missing jobs in the backoff cf, is more accurate imho

Reply:
@Zelldon can you clarify a bit further what you mean with your comment. I agree with @nicpuppa that his approach is more robust.

Do you mean that we should stop iterating over the failed jobs if we detect only 1 failed job more than the number of jobs in the backoff cf?

Reply:
Yes thanks or we just run it always, as there is no faster way to detect

Reply:
Since looking to just a failed job or run it always doesn't make too much difference, I will go with running the migration everytime, at least is a bit faster ⚡